pkg/cosign/keys.go

```diff
@@ -105,13 +182,42 @@ func PemToECDSAKey(pemBytes []byte) (*ecdsa.PublicKey, error) {
 	return ecdsaPub, nil
 }
 
+func LoadPrivateKey(key []byte, pass []byte) (signature.SignerVerifier, error) {
+
+	// Decrypt first
+	p, _ := pem.Decode(key)
+	if p == nil {
+		return nil, errors.New("invalid pem block")
+	}
+	if p.Type != PrivateKeyPemType {
+		return nil, fmt.Errorf("unsupported pem type: %s", p.Type)
+	}
+
+	x509Encoded, err := encrypted.Decrypt(p.Bytes, pass)
+	if err != nil {
+		return nil, errors.Wrap(err, "decrypt")
+	}
+
+	pk, err := x509.ParsePKCS8PrivateKey(x509Encoded)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing private key")
+	}
+	switch pk.(type) {
+	case *rsa.PrivateKey:
+		return LoadRSAPrivateKey(key, pass)
+	case *ecdsa.PrivateKey:
+		return LoadECDSAPrivateKey(key, pass)
+	}
```

Comment:
I wonder if we could, instead of having a fallthrough saying "loading private key" we can attempt to communicate whether it's an unsupported key type.

Reply:
Made changes.